modules/flowable-variable-service/src/main/java/org/flowable/variable/service/impl/AbstractVariableQueryImpl.java

```diff
@@ -225,7 +304,11 @@ protected void addVariable(String name, Object value, QueryOperator operator, bo
                 throw new FlowableIllegalArgumentException("Only string values can be used with 'like' condition");
             }
         }
-        queryVariableValues.add(new QueryVariableValue(name, value, operator, localScope));
+        if (scopeType != null)
+            queryVariableValues.add(new QueryVariableValue(name, value, operator, scopeType));
+        else
+            queryVariableValues.add(new QueryVariableValue(name, value, operator, localScope));
+
```

Comment:
Minor nitpick:  the project uses braces on `if/then/else` even when the body is a single statement.

Reply:
Thanks for your feedback. I added the brackets.